### PR TITLE
One click daily digest

### PR DESCRIPTION
**Pull request by bjfletcher**
Replace action link with an "in-page" button action, to sign up to the daily digest. The myFT code for the sign up is untested.

### CODE REVIEW

File: src/components/bottom/daily-digest/main.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+		function handleSignUpClick (evt) {
+			evt.preventDefault();
+			const conceptId = document.documentElement.dataset.conceptId;
+			if (conceptId) {
+				function addUserToConcept () {
+					return myft.add('user', null, 'followed', 'concept', conceptId);
+				}
+				function logError (err) {
+					/* eslint no-console:0 */
+					console.log({ info: 'could not add user to cocnept', conceptId, err });
```
Comment (CourtneyLO) [inline src/components/bottom/daily-digest/main.js]: I believe there is a typo here:
```suggestion
					console.log({ info: 'could not add user to concept', conceptId, err });
```

**Comment by bjfletcher** [comment]
@sentayhuJo I would do the following:

1. instead of generating HTML in JS, do it like this:

```
<div class="....front-banner ...--is-visible">...</div>
<div class="...back-banner">...</div>
```

Then when the subscription is successful, change `--is-visible` from `front-banner` to `back-banner` using JS & `classList`.

In this way, you can write all HTML in HTML, not HTML in JS. You can also do nice CSS transitions like flipping, should Emma want it!

2. instead of custom styling, always use Origami's styling:

https://registry.origami.ft.com/components/o-buttons@6.0.5
https://registry.origami.ft.com/components/o-icons@6.0.0 (see the "tick" one)
https://registry.origami.ft.com/components/o-colors@5.0.5 (if you're unsure which ones, ask Emma)

In this way, the styling, the look and feel will be consistent with the rest of FT.

**Comment by bjfletcher** [comment]
PS I've removed unused modules (`get-token` and `sign-up`)

File: secret-squirrel.js [hunk 2]
```diff
@@ -16,7 +16,8 @@ module.exports = {
 			'eb62aef2-4bdc-0f31-a2c5-6532b8e17896', // templates/partials/bottom/content-message.html:15
 			'5f60b8b4-cbf0-18d7-df41-9caa1171e8c1', // templates/partials/top/anon-subscribe-now.html:5|18
 			'a44c9005-2a9c-fe2a-9140-1311ff87f25f', // templates/partials/top/print-banner-usa.html:4
-			'79c8c0c5-1686-7f28-3aeb-c14688d59749' // templates/partials/top/uk-election.html:4
+			'79c8c0c5-1686-7f28-3aeb-c14688d59749', // templates/partials/top/uk-election.html:4
+			'82645c31-4426-4ef5-99c9-9df6e0940c00' // src/components/bottom/daily-digest/main.js: 9
```
Comment (tom-parker) [inline secret-squirrel.js]: I'm not sure this needs to be here

File: src/components/bottom/daily-digest/main.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
+import myft from 'next-myft-client';
+import Cookies from 'js-cookie';
+
+module.exports = function customSetup (banner, done) {
+	if (window.FT && window.FT.flags && window.FT.flags.oneClickDailyDigest) {
```
Comment (tom-parker) [inline src/components/bottom/daily-digest/main.js]: Do we definitely want to scope this message to a single flag? This custom JS setup should only be triggered if a user should be seeing this message so I don't think we need this double check?

File: secret-squirrel.js (hunk 2 above)
Comment (sentayhuJo) [reply secret-squirrel.js]: This offer Id is added by @remybach , https://github.com/Financial-Times/n-messaging-client/commit/03e830d16ebaecf3f10b927a1b60490a1f26203b

File: src/components/bottom/daily-digest/main.js (hunk 3 above)
Comment (saldixon) [reply src/components/bottom/daily-digest/main.js]: wasn't this purely for testing purposes?  this flag?  to be removed once we had confidence in the one-click behaviour?

File: secret-squirrel.js (hunk 2 above)
Comment (tom-parker) [reply secret-squirrel.js]: Yep, not the '82645c31-4426-4ef5-99c9-9df6e0940c00' value though. That's only appeared in this PR but I don't think it's actually used anywhere else in the code so we can probably remove it

File: src/components/bottom/daily-digest/main.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
+import myft from 'next-myft-client';
```
Comment (tom-parker) [inline src/components/bottom/daily-digest/main.js]: I don't think this file is needed for the `oneclick-daily-digest` messageID

File: src/components/bottom/daily-digest/main.scss [hunk 5]
```diff
@@ -0,0 +1,58 @@
+.o-banner_success-icon-tick {
```
Comment (tom-parker) [inline src/components/bottom/daily-digest/main.scss]: I don't think this file is needed for the `oneclick-daily-digest` messageID

File: templates/partials/bottom/daily-digest.html [hunk 6]
```diff
@@ -1,17 +1,31 @@
 <div class="n-messaging-client-messaging-banner">
```
Comment (tom-parker) [inline templates/partials/bottom/daily-digest.html]: There shouldn't be any changes to this file if we're using the new `oneclick-daily-digest` message ID

File: src/components/bottom/daily-digest/main.js (hunk 4 above)
Comment (sentayhuJo) [reply src/components/bottom/daily-digest/main.js]: did you mean for `daily-digest` message ID, coz originally we don't have such file for the daily digest.

File: src/components/bottom/daily-digest/main.scss (hunk 5 above)
Comment (sentayhuJo) [reply src/components/bottom/daily-digest/main.scss]: dito

**Comment by sentayhuJo** [comment]
This PR is close because we need to tidy up the change.

**Pull request closed without merging**